src/Microsoft.Health.Fhir.Shared.Api.UnitTests/Features/Formatters/HttpExtensionsTests.cs

```diff
@@ -0,0 +1,198 @@
+
+        public HttpExtensionsTests()
+        {
+            _logger = new NullLogger<string>();
+        }
+
+        [Fact]
+        public void GivenARequestWithSummaryType_WhenSerializingTheResponse_ThenTheCorrectSummeryTypeIsApplied()
+        {
+            var context = new DefaultHttpContext();
+            context.Request.QueryString = QueryString.Create("_summary", "text");
+
+            var summary = context.GetSummaryType(_logger);
+
+            Assert.Equal(SummaryType.Text, summary);
+        }
+
+        [Fact]
+        public void GivenARequestWithCapsSummaryType_WhenSerializingTheResponse_ThenTheCorrectSummeryTypeIsApplied()
+        {
+            var context = new DefaultHttpContext();
+            context.Request.QueryString = QueryString.Create("_SUMMARY", "DATA");
+
+            var summary = context.GetSummaryType(_logger);
+
+            Assert.Equal(SummaryType.Data, summary);
+        }
+
+        [Fact]
+        public void GivenARequestWithUnknownSummaryType_WhenSerializingTheResponse_ThenFalseIsReturned()
+        {
+            var context = new DefaultHttpContext();
+            context.Request.QueryString = QueryString.Create("_summary", "abc");
+
+            Assert.Throws<ArgumentException>(() => context.GetSummaryType(_logger));
+        }
+
+        [Fact]
+        public void GivenARequestWithNoSummaryType_WhenSerializingTheResponse_ThenFalseIsReturned()
+        {
+            var context = new DefaultHttpContext();
+
+            var summary = context.GetSummaryType(_logger);
+
+            Assert.Equal(SummaryType.False, summary);
+        }
+
+        [Fact]
+        public void GivenARequestWithPrettyIndentationSetToTrue_WhenSerializingTheResponse_ThenPrettyIndentationIsApplied()
+        {
+            var context = new DefaultHttpContext();
+            context.Request.QueryString = QueryString.Create("_pretty", "true");
+
+            var isPretty = context.GetIsPretty(_logger);
+
+            Assert.True(isPretty);
+        }
+
+        [Fact]
+        public void GivenARequestWithPrettyIndentationSetToFalse_WhenSerializingTheResponse_ThenPrettyIndentationIsNotApplied()
+        {
+            var context = new DefaultHttpContext();
+            context.Request.QueryString = QueryString.Create("_pretty", "false");
+
+            var isPretty = context.GetIsPretty(_logger);
+
+            Assert.False(isPretty);
+        }
+
+        [Fact]
+        public void GivenARequestWithPrettyIndentationSetToTrueInCaps_WhenSerializingTheResponse_ThenPrettyIndentationIsApplied()
+        {
+            var context = new DefaultHttpContext();
+            context.Request.QueryString = QueryString.Create("_PRETTY", "TRUE");
+
+            var isPretty = context.GetIsPretty(_logger);
+
+            Assert.True(isPretty);
+        }
+
+        [Fact]
+        public void GivenARequestWithPrettyIndentationSetToFalseInCaps_WhenSerializingTheResponse_ThenPrettyIndentationIsNotApplied()
```

Comment:
These are good tests. You could use xUnit's "theories" to parameterize the same test with different inputs and expected outputs to save some typing and space. Take a look at DecimalExtensionTests.cs

Reply:
Oh cool, thanks. Let me know what you think of 395dd9b.

Reply:
Looks good!